google/cloud/storage/examples/storage_bucket_samples.cc

```diff
@@ -221,6 +221,87 @@ void PatchBucketStorageClassWithBuilder(google::cloud::storage::Client client,
   (std::move(client), bucket_name, storage_class);
 }
 
+void AddBucketLabel(google::cloud::storage::Client client, int& argc,
+                    char* argv[]) {
+  if (argc != 4) {
+    throw Usage{"add-bucket-label <bucket-name> <label-key> <label-value>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  auto label_key = ConsumeArg(argc, argv);
+  auto label_value = ConsumeArg(argc, argv);
+  //! [add bucket label] [START storage_add_bucket_label]
+  namespace gcs = google::cloud::storage;
+  [](gcs::Client client, std::string bucket_name, std::string label_key,
+     std::string label_value) {
+    gcs::BucketMetadata updated_metadata = client.PatchBucket(
+        bucket_name,
+        gcs::BucketMetadataPatchBuilder().SetLabel(label_key, label_value));
+    std::cout << "Successfully set label " << label_key << " to " << label_value
+              << " on bucket  " << bucket_name << ".";
+    std::cout << " The bucket labels are now:";
+    for (auto const& kv : updated_metadata.labels()) {
+      std::cout << "\n  " << kv.first << ": " << kv.second;
+    }
+    std::cout << std::endl;
+  }
+  //! [add bucket label] [END storage_add_bucket_label]
+  (std::move(client), bucket_name, label_key, label_value);
+}
+
+void GetBucketLabels(google::cloud::storage::Client client, int& argc,
+                     char* argv[]) {
+  if (argc != 2) {
+    throw Usage{"get-bucket-labels <bucket-name>"};
+  }
+  auto bucket_name = ConsumeArg(argc, argv);
+  //! [get bucket labels] [START storage_get_bucket_labels]
+  namespace gcs = google::cloud::storage;
+  [](gcs::Client client, std::string bucket_name) {
+    gcs::BucketMetadata metadata =
+        client.GetBucketMetadata(bucket_name, gcs::Fields("labels"));
+    if (metadata.label_count() == 0U) {
```

Comment:
What about `metadata.has_labels()`?

Reply:
We have `has_label()`, but no `has_labels()`: I am happy to add it though.
I thought we liked to avoid chaining? But if we are using chaining, `metadata.labels().empty()` would be better in this case.

I am not sure whether you want a change here, and if so which change?

If you want my opinion, I think we should go with fewer convenience functions, even `label_count()` seems like a bad idea in hindsight.

Reply:
Thanks for the feedback. I thought about your feedback on `for()` using `auto` versus not and then I re-reviewed this sample.

Assumption: I don't think developers will count the number of labels or check they're the map is empty. I think they would likely iterate through them.

In comparison, `ObjectMetadata::upsert_metadata()`, made sense to me as a convenience method because a user would need to method chain every time they wanted to update/insert user metadata. The convenience method reduces that, but there's always an alternative `ObjectMetadata::mutable_metadata()`.

One other bit is BucketMetadata is missing a similar `upsert_metadata()` method for labels. Is this by design?

Reply:
Sounds like we should not use `label_count()` in the examples, and we should remove the function from `ObjectMetadat` and `BucketMetadata`.  I agree, I do think that removing the functions should go into another PR, I created #1148 to track the removal.

> One other bit is BucketMetadata is missing a similar upsert_metadata() method for labels. Is this by design?

That is #1046.

Reply:
Apologies for the confusion, you're correct. Thank you.